src/kern/npf_tableset.c

```diff
@@ -456,13 +456,18 @@ npf_table_getid(npf_table_t *t)
  * npf_table_check: validate the name, ID and type.
  */
 int
-npf_table_check(npf_tableset_t *ts, const char *name, uint64_t tid, uint64_t type)
+npf_table_check(npf_tableset_t *ts, const char *name, uint64_t tid, uint64_t type, bool replacing)
 {
+	const npf_table_t *t;
 	if (tid >= ts->ts_nitems) {
 		return EINVAL;
 	}
-	if (ts->ts_map[tid] != NULL) {
-		return EEXIST;
+	if ((t = ts->ts_map[tid]) != NULL) {
+		if (!replacing) {
+			return EEXIST;
+		} else if (strcmp(t->t_name, name) != 0) {
```

Comment:
Generally, I think there is no reason to prohibit the table name change. So, this could be removed.. however, in such case, the `npf_tableset_getbyname()` test below would need to be improved to handle the case when there is another table with a duplicate name and a different ID.

Reply:
I was a bit concerned that the table is likely to be referenced from rules in the running config? What are the consequences of a referenced table name going missing?

Reply:
I've had a brief look at this and have discovered that table names appear to be resolved to IDs at load time.
I'll have a look at implementing the change you've suggested.

Reply:
Would you be happy with this:

```c
	if ((t = ts->ts_map[tid]) != NULL && !replacing) {
		return EEXIST;
	}
```

and change the npf_tableset_getbyname() test to:

```c
	if ((dup_t = npf_tableset_getbyname(ts, name)) != NULL) {
		if (!replacing || dup_t->t_id != t->t_id) {
			return EEXIST;
		}
	}
```
?

Reply:
Yes, as you discovered, names are resolved to IDs on load. In the case of table rename, the subsequent `npfctl reload` would discard such tables if their name would not appear in the config. However, in terms of libnpf API, there is no reason to prohibit renaming, since not everybody might use `npfctl`. In fact, there is a client using only libnpf and they might even find this feature useful.

I think your suggested change is good. Just s/dup_t/dt/, otherwise it looks like a type ;) You will also need to handle the `t != NULL` case in the check.

Reply:
No problem, agreed about the `_t` variable suffix (not sure how I missed that! :o ) and I've changed the `t->t_id` reference to use the `tid` function argument instead.